extensions/amp-form/0.1/amp-form.js

```diff
@@ -1223,6 +1229,80 @@ export class AmpForm {
     }
   }
 
+  /**
+   * Initialize form fields from query parameter values if attribute
+   * 'data-initialize-from-url' is present on the form and attribute
+   * 'data-allow-initialization' is present on the field.
+   * @private
+   */
+  maybeInitializeFromUrl_() {
+    if (
+      isProxyOrigin(this.win_.location) ||
+      !this.form_.hasAttribute('data-initialize-from-url')
+    ) {
+      return;
+    }
+
+    const valueTags = ['SELECT', 'TEXTAREA'];
+    const valueInputTypes = [
+      'color',
+      'date',
+      'datetime-local',
+      'email',
+      'hidden',
+      'month',
+      'number',
+      'range',
+      'search',
+      'tel',
+      'text',
+      'time',
+      'url',
+      'week',
+    ];
+    const checkedInputTypes = ['checkbox', 'radio'];
+
+    const queryParams = parseQueryString(this.win_.location.search);
+    Object.keys(queryParams).forEach(key => {
+      const field = this.form_./*OK*/ querySelector(`[name="${key}"]`);
+      if (!field) {
+        return;
+      }
+      // Do not interfere with form fields that utilize variable substitutions.
+      // These fields are populated at time of form submission.
+      if (field.hasAttribute('data-amp-replace')) {
+        return;
+      }
+      // Form fields must be whitelisted
+      if (!field.hasAttribute('data-allow-initialization')) {
+        return;
+      }
+
+      const value = queryParams[key] || '';
+      const type = field.getAttribute('type') || 'text';
+      const tag = field.tagName;
+
+      if (
+        (tag === 'INPUT' &&
+          valueInputTypes.includes(type.toLocaleLowerCase())) ||
+        valueTags.includes(tag)
+      ) {
+        if (field.value !== value) {
+          field.value = value;
+        }
+      } else if (tag === 'INPUT' && checkedInputTypes.includes(type)) {
```

Comment:
Nit: Extract duplicate `tag === 'INPUT'` check into an outer conditional.

Reply:
This should be resolved in 1894b76.